src/wallet/walletdb.cpp

```diff
@@ -529,6 +534,13 @@ DBErrors WalletBatch::LoadWallet(CWallet* pwallet)
     if (wss.fAnyUnordered)
         result = pwallet->ReorderTransactions();
 
+    // Upgrade all of the wallet keymetadata to have the hd master key id
+    try {
```

Comment:
Nit: add comment that this is not atomic, but that individually updated keymetadata entries don't prevent older wallets from loading.

Reply:
Done